crates/polars-arrow/src/legacy/kernels/take_agg/mod.rs

```diff
@@ -12,7 +12,7 @@ use crate::types::NativeType;
 
 /// Take kernel for single chunk without nulls and an iterator as index.
 /// # Safety
-/// caller must ensure iterators indexes are in bounds
+/// caller must ensure iterators are not empty and indexes are in bounds
```

Comment:
It's fine if the iterator is empty though? Doesn't the function just return None then?